### PR TITLE
Add Component Spec to dependencies

### PR DESCRIPTION
**Pull request by AndreasBurger**
**What this PR does / why we need it**:
adds gardener component spec to our component-dependencies.

### CODE REVIEW

File: .ci/component_descriptor [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        )):
+            raise RuntimeError('Unable to find component-spec in requirements.txt')
+        if len(component_spec_requirement.specs) > 1:
+            raise RuntimeError('No exact version specified for component-spec dependency')
+        component_spec_specs = component_spec_requirement.specs[0]
+
+        return component_spec_specs[1]
+
+
+def add_dependence_on_component(
```
Comment (ccwienk) [inline .ci/component_descriptor:58]: dependencY

File: .ci/component_descriptor [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            version=component_version,
+            labels=[],
+        )
+    )
+
+
+component_descriptor = parse_component_descriptor()
+component = component_descriptor.component
+
+add_dependence_on_component(
```
Comment (ccwienk) [inline .ci/component_descriptor:78]: I think this helper function would not rm a lot of redundancy, even if it were used more than once. Like this, is merely adds boilerplate imo

File: .ci/component_descriptor [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    component=component,
+    component_name='github.com/gardener/component-spec',
+    component_version=f'v{component_spec_version()}',
```
Comment (ccwienk) [inline .ci/component_descriptor:81]: I think we should simply add it into the requirements-file with a v-prefix as well; according to setuptools, it is acceptable / semantically equivalent to do so (although not considered a good practice)

File: .ci/set_dependency_version [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python3
+
+import fileinput
+import os
+
+import util
+
+COMPONENT_SPEC_PROJECT_NAME = 'gardener-component-model'
```
Comment (ccwienk) [inline .ci/set_dependency_version:8]: deduplidate?

File: .ci/set_dependency_version [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        else:
+            print(line)
+
+
+util.check_env('REPO_DIR')
+
+component_name = util.check_env('DEPENDENCY_NAME')
+component_version = util.check_env('DEPENDENCY_VERSION')
+
+if component_name == 'github.com/gardener/component-spec':
```
Comment (ccwienk) [inline .ci/set_dependency_version:28]: deduplicate

File: .ci/set_dependency_version [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    update_project_version(COMPONENT_SPEC_PROJECT_NAME, component_version)
+else:
+    raise NotImplementedError(f"Don't know how to upgrade '{component_name}'")
```
Comment (ccwienk) [inline .ci/set_dependency_version:31]: nit: please always append `=` in format-str-expressions, unless there is a reason against doing so

File: .ci/set_dependency_version [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+            'requirements.txt',
+        )
+    for line in fileinput.input(path_to_requirements_txt, inplace=True):
+        if line.startswith(project_name):
+            print(f'{project_name}=={project_version}')
+        else:
+            print(line)
+
+
+util.check_env('REPO_DIR')
```
Comment (ccwienk) [inline .ci/set_dependency_version:23]: why use both MAIN_REPO_DIR and REPO_DIR?

**Review by ccwienk — CHANGES_REQUESTED** [review]

File: .ci/component_descriptor (hunk 1 above)
Comment (AndreasBurger) [reply .ci/component_descriptor:58]: that was intentional. Will change it to `add_component_dependency`

**Pull request merged**